### DISCUSSION TITLE
MTC support - convert Midi Clock to Midi Time Code

### DISCUSSION THREAD

**Discussion by cyberluke** [tttapa/Control-Surface#765] [Q&A] [answered]
Hi, I'm trying to solve iPad Cubasis to PC/Mac Studio One sync. One is using Midi Clock (sending a position as well) while the other is receiving only Midi Time Code MTC. Is there a support for MTC in this library? I would like to convert Midi Clock data to Time Code data: https://www.reddit.com/r/ipadmusic/comments/voj799/ipad_sync_to_studio_one_mac_only_is_ok_win_is/

**Comment by tttapa** [comment, marked as answer]
There is support in the sense that Control Surface allows you to send and receive MTC messages, using the [sendMTCQuarterFrame()](https://tttapa.github.io/Control-Surface-doc/Doxygen/d6/d72/classMIDI__Sender.html#a2898b25638c21b41b2cba1b656e73303) and [sendSysEx()](https://tttapa.github.io/Control-Surface-doc/Doxygen/d6/d72/classMIDI__Sender.html#a0ec7247d0c9a496856da5f788d8623da) functions for sending, and the appropriate [callbacks](https://tttapa.github.io/Control-Surface-doc/Doxygen/da/d27/MIDI-Input-Fine-Grained-All-Callbacks_8ino-example.html) when receiving.

**Reply by cyberluke** [reply]
Perfect, that should be enough. I see ESP32 and Teensy are supported. That will be my aim. The plan is to synchronize two DAWs, like old tape style. I need to explore how to decode MTC format. 

Documentation is great, but if there would be some example, it would be even better. 

The best use case would be if it would be possible to use USB hub on output USB port and send or receive two devices over usb hub. It worked for me on Teensy with keyboard and mouse (usb hid protocol). Also I will have USB host feature.

I build my own devices as hobbyist. Currently, my favourite platform is M5 Stack as you can stack modules without soldering. It is ESP32 based solution and have USB host module.
